subsys/net/ip/net_core.c

```diff
@@ -403,6 +403,190 @@ int net_send_data(struct net_pkt *pkt)
 	return 0;
 }
 
+#if defined(CONFIG_NET_PROMISCUOUS_MODE)
+static bool promiscuous_mode_enabled;
+
+K_FIFO_DEFINE(promiscuous_mode_queue);
+
+NET_PKT_SLAB_DEFINE(promiscuous_mode_pkts,
+		    CONFIG_NET_PROMISCUOUS_MODE_PKT_COUNT);
+NET_PKT_DATA_POOL_DEFINE(promiscuous_mode_bufs,
+			 CONFIG_NET_PROMISCUOUS_MODE_BUF_COUNT);
+
+#ifdef CONFIG_NET_L2_IEEE802154
```

Comment:
I really don't like that: mixing L2 stuff into net core. Big no no.

I think this will require a change in l2 API, so that core can request a generic hw setting to the L2 layer.

Reply:
Heh, I was planning originally to do just that but knowing your opinion about extending L2 API I went this route instead.

Reply:
Extending API is fine as long as it's a generic change that will benefit all L2s.

Actually, I am not entirely sure we need to do promiscuous stuff via net_if. It's a very specific feature, for even more specific use cases. So we may just stick to net_mgmt for that.

Yeah I think a net if net mgmt for it would be better: it would not bloat L2 API plus it could be easily conditioned not to be built via one option, no ifdefs anywhere etc...

Idea would be:

- a "generic" net if net_mgmt interface promiscuous call
- below it would be a "gateway" to L2 mgmt: it would check the iface type, if ethernet --> it would call the right ethernet net mgmt for it, same for 15.4 (there is none yet) etc...

It bloats a tiny bit due to the requirement of net_mgmt, but it's ok because it's not going to be used on too constrained devices and only for highly specific use-case.

The only requirement in net_if side it to be able to set/unset promiscuous flag (such functions would be private to net stack obviously).

Reply:
Hmm, I am not sure about the net_mgmt route here.
Anyway, I will send the generalized net_l2 version as I already converted the PR to use that.